flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java

```diff
@@ -52,22 +52,32 @@ public class ForStStateExecutor implements StateExecutor {
      */
     private final ExecutorService coordinatorThread;
 
-    /** The worker thread that actually executes the {@link StateRequest}s. */
-    private final ExecutorService workerThreads;
+    /** The worker thread that actually executes the read {@link StateRequest}s. */
+    private final ExecutorService readThreads;
+
+    /** The worker thread that actually executes the write {@link StateRequest}s. */
+    private final ExecutorService writeThreads;
 
     private final RocksDB db;
 
     private final WriteOptions writeOptions;
 
     private Throwable executionError;
 
-    public ForStStateExecutor(int ioParallelism, RocksDB db, WriteOptions writeOptions) {
+    public ForStStateExecutor(
+            int readIoParallelism, int writeIoParallelism, RocksDB db, WriteOptions writeOptions) {
```

Comment:
How about also supporting merged thread pool for R/W? e.g.` if (readIoParallelism <= 0 || writeIoParallelism <= 0) `then `readThreads` = `writeThreads` = a thread pool with `Math.max(readIoParallelism, writeIoParallelism)` threads?

Also a sanity check where `readIoParallelism` and `writeIoParallelism` should not be less than 0 at the same time.

Reply:
👍Thanks for the suggestion, fixed